### PR TITLE
[ADD] use browse method to pull the forecast cache data instead of make a read() call. The read call was returning an error when making the calculate of the forecast function fields, an access error.

### LINKED ISSUE #36: Fix security issues in forecasting_smoothing_techniques module
When trying to create a new forecast record show en error indicating that the user has not read permissions.

image

TODO

- [x] config runbot an test if this issue can be repeat at runbot.
- [ ] extend the unit test about security in the forecasting_smoothing_techniques module.
- [x] ask @moylop260 for help to reproduce the error.

### PR DESCRIPTION
**Pull request by zaoral**
Fix #36
Task [vx#4163](https://www.vauxoo.com/web#id=4163&view_type=form&model=project.task&action=138) HU [913](https://www.vauxoo.com/web#id=913&view_type=form&model=user.story&action=726) Sprint 20

### CODE REVIEW

**Comment by zaoral** [comment]
@nhomar This PR is ready to review

File: forecasting_smoothing_techniques/models/forecast.py [hunk 1]
```diff
@@ -335,7 +338,7 @@ def almost_equal(self, actual, expected):
         """
         vdiff = abs(actual - float(expected))
         allowed_error = 2
-        if int(round(vdiff)) > allowed_error:
+        if int(round(vdiff)) <= allowed_error:
```
Comment (nhomar) [inline forecasting_smoothing_techniques/models/forecast.py]: esto está bien? cambiaste completamente el sentido de la igualdad

File: forecasting_smoothing_techniques/tests/test_forecasting.py [hunk 2]
```diff
@@ -492,7 +499,21 @@ def test_06_2(self):
...
+        forecast = self.forecast_obj.search([])[0]
+        forecast.sudo(forecast_user).write({
+            'name': 'New name was set in unit test 06 2'})
+        self.assertTrue(forecast.sudo(forecast_user).name)
+        forecast2 = forecast.sudo(forecast_user).copy()
+        self.assertTrue(forecast2)
+        # TODO this is not properly working. show a integrity error and must
+        # show a accessError
+        # with self.assertRaises(Exception):
+        #     forecast.sudo(forecast_user).unlink()
```
Comment (nhomar) [inline forecasting_smoothing_techniques/tests/test_forecasting.py]: no dejes líneas comentadas que no necesitas.

File: forecasting_smoothing_techniques/models/forecast.py (hunk 1 above)
Comment (zaoral) [inline forecasting_smoothing_techniques/models/forecast.py]: Yes, almost_equal method was making the inverse checking, here, and in the unit test. The logic now is correct in both elements and know works properly.

**Comment by zaoral** [comment]
@nhomar I update the PR with the changes you request

**Pull request merged**